File: Seção 10 - Linguagem C++/Exercicios/Exercicio_7.cpp
#include <iostream>
using namespace std;


int main(){

    int M;

    cout << "Qual a ordem da matriz? ";
    cin >> M;

    int i, j;
    int mat[M][M];

    for (i = 0; i < M; i++){
        for (j = 0; j < M; j++){
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
        }
    }



    cout << "DIAGONAL PRINCIPAL: " << endl;
    for (i = 0; i < M; i++){
        cout << mat[i][i] << " ";
    }

    cout << endl;
    int somaneg;
    somaneg = 0;

    for (i = 0; i < M; i++){
        for (j = 0; j < M; j++){
            if (mat[i][j] < 0){
                somaneg = somaneg + 1;
            }
        }
    }

    cout << "QUANTIDADE DE NEGATIVOS = " << somaneg;

return 0;
}

File: Seção 11 - Linguagem Java/Exercícios/Exercicio_7.java
import java.util.Scanner;

public class Exercicio_7 {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
				
		System.out.print("Qual a ordem da matriz? ");
		int N = sc.nextInt();
		int[][] mat = new int[N][N];
		
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				System.out.print("Elemento [" + i + "," + j + "]: ");
				mat[i][j] = sc.nextInt();
			}
		}
		
		System.out.println("DIAGONAL PRINCIPAL = ");
		
		for (int i = 0; i < N; i++) {
			System.out.print(mat[i][i] + " ");
		}
		
		int somaneg = 0;
		
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				if (mat[i][j] < 0) {
					somaneg = somaneg + 1;
				}
			}
		}
		
		System.out.println();
		
		System.out.print("QUANTIDADE NEGATIVOS = " + somaneg);
		
		sc.close();
	}

}
